feat!: made `deployContract` a non-blocking call (#2597)
FuelLabs/fuels-ts: apps/docs/src/guide/contracts/deploying-contracts.md

## 4. Deploying the Contract

> To deploy the contract, instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Then, call the `deployContract` method. This method resolves as soon as the transaction to deploy the contract is submitted and returns two objects: a `TransactionResponse` and a `Contract` instance.

Reviewer: ```suggestion
To deploy the contract, instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Then, call the `deployContract` method. This method resolves when the transaction to deploy the contract is submitted, returning two objects: a `TransactionResponse` and a `Contract` instance.
```

<<< @/../../docs-snippets/src/guide/contracts/deploying-contracts.test.ts#contract-setup-3{ts:line-numbers}

> **Important**: The contract instance can only be safely used after the promise from `transactionResponse.waitForResult` is resolved. To avoid blocking your code, attach this promise to a hook or listener that will use the contract only after it is fully deployed.

Reviewer: ```suggestion
**Important**: The contract instance can only be safely used after the promise from `transactionResponse.waitForResult` is resolved. To avoid blocking your code, attach this promise to a hook or listener that awaits it to ensure you don't use the contract before its deployment finishes.
```

## 4. Deploying the Contract

> To deploy the contract we can instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. The we can call `deployContract` method. This method returns a promise that will resolve to a `Contract` instance, ready to be used to interact with the deployed smart contract.

Reviewer 2: ```suggestion
To deploy the contract we can instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Then we can call the `deployContract` method. This method returns a promise that will resolve to a `Contract` instance which provides methods to interact with the deployed smart contract.
```

Author: @danielbate The docs are slightly different now https://github.com/FuelLabs/fuels-ts/pull/2597/commits/7c800e1786393467ffb92433a7a331d6ca38d4cb

<<< @/../../docs-snippets/src/guide/contracts/deploying-contracts.test.ts#contract-setup-3{ts:line-numbers}

The `deployContract` method will only resolve after the deploy contract transaction has been processed and the smart contract is deployed on the blockchain. As a result, this call can take several seconds to resolve and will block any code execution that follows.

If this situation is inappropriate for your dApp, you can use the `deployContractAsync` method.

## 5. Deploying the Contract Asynchronously

> In some cases, you may not want to wait for the deploy contract transaction to finish processing. In these instances, you can use `deployContractAsync`. This method resolves as soon as the transaction to deploy the contract is submitted and returns two objects: a `TransactionResponse` and a `Contract` instance.

Reviewer 2: I'd probably make the point about the `waitForResult` here as that seems like the most important point here, and the key distinction between the 2 methods.

Author: @danielbate The docs are slightly different now https://github.com/FuelLabs/fuels-ts/pull/2597/commits/7c800e1786393467ffb92433a7a331d6ca38d4cb

## 4. Deploying the Contract

> To deploy the contract, instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Then, call the `deployContract` method. This call resolves as soon as the transaction to deploy the contract is submitted and returns two items: the `transactionId` and a `waitForResult` function.

Reviewer 3: ```suggestion
To deploy the contract, instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Then, call the `deployContract` method. This call resolves as soon as the transaction to deploy the contract is submitted and returns three items: the `contractId`, the `transactionId` and a `waitForResult` function.
```
